.circleci/config.yml

```diff
@@ -209,104 +329,150 @@ jobs:
-            git push origin v${GUS_BUILD}
+            git tag v${RELEASE_VERSION}
+            git push origin v${RELEASE_VERSION}
             git push origin master
       - sfchangecase/update:
           location: REPO_LOCATION
           release: GUS_RELEASE
+      - slack/notify:
+          channel: 'pdt_releases'
+          color: '#9bcd9b'
+          message: 'VSCode Extensions v${RELEASE_VERSION} have been published to the marketplace'
+      - slack/status:
+          channel: 'pdt_releases'
+          fail_only: true
 
   create-release-branch:
-    <<: *defaults
+    parameters:
+      e:
+        type: executor
+    executor: << parameters.e >>
     steps:
-      - add_ssh_keys: *ssh-config
+      - slack/notify:
+          channel: 'pdt_releases'
+          color: '#FFDEAD'
+          message: 'Starting creation of release branch for VSCode Extensions'
+      - ssh-config
       - checkout
-      - restore_cache: *restore_cache
-      - run: *gh-config
-      - run: *install
+      - restore_dependency_cache
+      - gh-config
+      - install
       - run:
           name: Install Lerna
-          command: sudo npm install -g lerna@3.13.1
+          command: sudo npm install -g lerna@<< pipeline.parameters.lerna-version >>
       - run:
           name: Create release branch
           command: |
             git branch
             scripts/create-release-branch.js
+      - slack/notify:
+          channel: 'pdt_releases'
+          color: '#d8bfd8'
+          message: 'VSCode Extensions release branch has been cut'
+      - slack/status:
+          channel: 'pdt_releases'
+          fail_only: true
 
 workflows:
   version: 2.1
 
   # run a build and a windows test on every commit
   commit-workflow:
+    when:
+      and:
+        - not: << pipeline.parameters.publish >>
+        - not: << pipeline.parameters.create-release >>
     jobs:
       - build-all
       - unit-tests
       - integration-tests
+
+  publish-workflow:
+    when: << pipeline.parameters.publish >>
+    jobs:
+      - slack/approval-notification:
+          message: Pending Approval for Publish
+          channel: 'pdt_releases'
+          color: '#0E1111'
+      - build-all
+      - unit-tests
+      - integration-tests
+      - hold: # Requires manual approval in Circle Ci
+          type: approval
       - publish:
+          e: default-executor
+          requires:
+            - build-all
+            - unit-tests
+            - integration-tests
+            - hold
           filters:
             branches:
               only:
                 - master
+
+  create-release-branch-workflow:
+    when: << pipeline.parameters.create-release >>
+    jobs:
+      - slack/approval-notification:
+          message: Pending Approval for Creation of Release Branch
+          channel: 'pdt_releases'
+          color: '#0E1111'
+      - build-all
+      - unit-tests
+      - integration-tests
+      - hold: # Requires manual approval in Circle Ci
```

Comment:
@sfsholden - Good job on this. Luis asked me to take a look and things look 👍 

Quick question on the approval.

Can I approve my own workflow? Or must it be a different individual?

Reply:
@vazexqi 
Thanks for taking a look!

Anyone with push access to the repo by default will be able to approve the workflow. [Here](https://circleci.com/docs/2.0/workflows/#holding-a-workflow-for-a-manual-approval) is more info on what goes into the approval process.

To restrict the approval process further you can add [contexts](https://circleci.com/docs/2.0/contexts/#approving-jobs-that-use-restricted-contexts).